src/librustc_codegen_utils/symbol_names.rs

```diff
@@ -184,7 +184,7 @@ fn symbol_name(tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> InternedString {
 
     let instantiating_crate = if avoid_cross_crate_conflicts {
         Some(if is_generic {
-            if !def_id.is_local() && tcx.sess.opts.share_generics() {
+            if !def_id.is_local() && tcx.share_generics(def_id.krate) {
```

Comment:
It seems worthwhile to change all the use sites to the query instead of sess.opts and perhaps rename the method on sess.opts to `share_generics_untracked`.

Reply:
Good observation, @Mark-Simulacrum! I would even go a step further and remove `sess.opt.share_generics()` in favor of the local query provider.